treesapp_assign_and_update.Rmd

```diff
@@ -72,31 +89,58 @@ do
+From there, it will figure out what sequences were classified and what their assigned lineages were.
+
+By default TreeSAPP will use the lineage that was assigned to each classified query sequence from the `treesapp assign` outputs.
+Although these sequences are from UniProt (and originally GenBank) they do not have NCBI accessions or taxonomy IDs in the header, so we must provide taxonomic lineages for each sequence in a table.
+Fortunately, we already have this information in the files `UniProt_AmoA.tab` and `UniProt_PmoA.tab`.
+
+- Create a table mapping sequence names to taxonomic lineage information for `treesapp update`.
+The first command creates a new file (`UniProt_XmoA.tsv`) with the table column names.
+The second command pipes all rows in the UniProt lineage tables except the header line by using the `tail` command.
+The `-q` parameter prevents `tail` from including the file name in the output.
+The UniProt table rows are appended to the new file by using `>>` instead of one `>`.
+
+```{bash eval = FALSE}
+echo -e "SeqID\tOrganism\tDomain\tPhylum\tClass\tOrder\tFamily\tGenus\tSpecies" >UniProt_XmoA.tsv
+tail -q -n +2 UniProt_*tab >>UniProt_XmoA.tsv
+```
+
+Alternatively, inverse pattern matching the word "Entry" --- the first column name in the exported UniProt tables --- with `grep --no-filename -v Entry UniProt_*tab` could have also worked.
 
-By default TreeSAPP will use the lineage that was assigned to each classified query sequence from the `treesapp assign` outputs. Since these sequences are from FunGene (and originally GenBank) they have NCBI accessions, meaning we can use this information to download their true taxonomic lineages. To turn on that behaviour we use the `--skip_assign` flag.\
-Lastly, we must give it the path to the reference package to update with `--refpkg_path XmoA_seed/final_outputs/XmoA_build.pkl`.
+To make TreeSAPP use this new table instead of using the taxonomic classifications generated by `treesapp assign` we use the `--skip_assign` flag.\
+Finally, we are going to require that taxonomic lineages are resolved to at least the rank of *Order* with the argument `--min_taxonomic_rank`.
+Resolving lineages of sequences to at least this rank will help us link them to their respective CuMMO groups more precisely.
+This level of resolution isn't necessary for all reference packages but given the metabolic diversity within the CuMMOs of *Proteobacteria* alone, it is here.
 
-__NOTE:__ `--fastx_input XmoA_seed.faa` has been removed from all `treesapp update` commands as of TreeSAPP 0.10.4 being installed on the server. See [above](#change-notes) for more details.
+- Update the reference package `XmoA_seed/final_outputs/XmoA_build.pkl` with classified UniProt XmoA sequences.
+Use the argument `--seqs2lineage` to specify the path to the table containing taxonomic lineages for all UniProt XmoA sequences.
 
 ```{bash eval = FALSE}
 treesapp update \
```

Comment:
Minor
This one took a long time to run and there were times when it just looked like it was stuck.

Reply:
Good to know. I'll rerun the tutorial on the server after merging. Hopefully it was caused by an intermittent network issue. TreeSAPP may query the Entrez database when compiling lineage information and that step can be slow.